### DISCUSSION TITLE
If I know the id of the vector, can I know its code and the key of the inverted list it is in?

### DISCUSSION THREAD

**Discussion by Learningisboundless** [facebookresearch/faiss#3668] [Q&A]
In the IVFPQ search process, does the key value of the coarse centroid correspond to the list_no in the invertedlists? If I know the id of the vector in the dataset, can I get its code and the key of the invertedlist where it is located? Thanks for your help !
running on cpu
interface is c++

**Comment by mdouze** [comment]
yes, if you enable direct map, see https://github.com/facebookresearch/faiss/wiki/Special-operations-on-indexes#direct-map-for-an-indexivf

In that case, the inverted list for a vector id is
```c++
index.set_direct_map_type(DirectMap.Array)

idx_t vector_id = ....; 
idx_t list_no = lo_listno(index.direct_map.get(vector_id));

```

**Comment by Learningisboundless** [comment]
After getting list_no, can I know the offset of the vector's code in the inverted list? So I can get the vector's code. Or is there any other way I can get the code for the vector?
